tools/tplg_parser/include/tplg_parser/topology.h

```diff
@@ -38,11 +39,52 @@ struct sof;
 struct fuzz;
 struct sof_topology_module_desc;
 
+struct sof_ipc4_audio_format {
+	uint32_t sampling_frequency;
+	uint32_t bit_depth;
+	uint32_t ch_map;
+	uint32_t ch_cfg; /* sof_ipc4_channel_config */
+	uint32_t interleaving_style;
+	uint32_t fmt_cfg; /* channels_count valid_bit_depth s_type */
+} __packed __aligned(4);
+
+/**
+ * struct sof_ipc4_pin_format - Module pin format
+ * @pin_index: pin index
+ * @buffer_size: buffer size in bytes
+ * @audio_fmt: audio format for the pin
+ *
+ * This structure can be used for both output or input pins and the pin_index is relative to the
+ * pin type i.e output/input pin
+ */
+struct sof_ipc4_pin_format {
+	uint32_t pin_index;
+	uint32_t buffer_size;
+	struct sof_ipc4_audio_format audio_fmt;
+};
```

Comment:
@ranj063 audio_format is packed but not this one, is this safe?

Reply:
@kv2019i yes it is OK.  this struct is only used to parse and save the topology audio formats info. It is never sent in the IPC payload so it doesnt have to be packed.